src/promptflow/promptflow/_sdk/_service/utils.py

```diff
@@ -20,8 +22,26 @@ def wrapper(*args, **kwargs):
             response.status_code = 404
             return response
         except Exception:  # pylint: disable=broad-except
-            response = jsonify({"error_message": "Internal Server Error"})
+            response = jsonify({"error_message": f"Internal Server Error"})
             response.status_code = 500
             return response
 
     return wrapper
+
+
+def validate_request_user():
+    # Get the user name from request.
+    user = request.environ.get("REMOTE_USER") or request.headers.get("X-Remote-User")
+    if user != getpass.getuser():
+        abort(403)
```

Comment:
what's the error message user will get in this case?

Reply:
Response of invalid user
![image](https://github.com/microsoft/promptflow/assets/17938940/1bdb156b-43ef-4f17-bfb9-2f5b96bce5e5)